peps/pep-0776.rst

```diff
@@ -0,0 +1,794 @@
+
+About 60 packages are also testing against Pyodide in their CI, including NumPy,
+pandas, awkward-cpp, scikit-image, statsmodels, PyArrow, Hypothesis, and PyO3.
+
+Emscripten Wheel Format
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Emscripten wheels will use either the format ``emscripten_<version>_wasm32`` or
+``pyodide_<abi>_wasm32``. For example:
+
+* ``emscripten_3_1_58_wasm32``
+* ``pyodide_2025_0_wasm32``
+
+The first triple is ambiguous, since even with Emscripten 3.1.58 it is possible
+to link dynamic libraries that require a large number of distinct ABIs,
+depending on linker and compiler options. It is our intent that the
+``pyodide_2025_0`` specifies the particular ABI. Thus, the relationship between
+``pyodide_<abi>`` and ``emscripten_<version>`` is intended to be the same as the
+relationship between ``manylinux<version>`` and ``linux``.
+
+The specification of the ``pyodide_<abi>`` ABI includes:
+
+* Which version of the Emscripten compiler is used
+* What libraries are statically linked with the interpreter
+* What stack unwinding ABI is to be used
+* Which runtime platform features are required to be present
+
+and a handful of other similar details that affect the ABI.
+
+The ABI is selected by choosing the appropriate version of the Emscripten
+compiler and passing appropriate compiler and linker flags. It is possible for
+other people to build their own Python interpreter that is compatible with the
+Pyodide ABI, it is not necessary to use the Pyodide distribution itself.
+
+The ``pyodide build`` tool knows how to create wheels that match our ABI. As an
+alternative,
+`the auditwheel-emscripten tool <https://github.com/ryanking13/auditwheel-emscripten>`__
+is capable of performing basic compatibility checks, vendoring shared libraries,
+and retagging the wheel from ``emscripten_<version>`` to ``pyodide_<abi>``. Unlike
+with manylinux, there is no need for a Docker container to build the
+``pyodide_<abi>`` wheels. All that is needed is a Linux machine and appropriate
+versions of Python, Node.js, and Emscripten.
+
+
+PEP 11
+------
+
+:pep:`11` will be updated to indicate that Emscripten is supported, specifically
+the triples ``wasm32-unknown-emscripten_xx_xx_xx``.
+
+Russell Keith-Magee will serve as the initial core team contact for these ABIs.
+
+
+Future Work
+===========
+
+Improving Cross Builds in the Packaging Ecosystem
+-------------------------------------------------
+
+Python now supports four non-self-hosting platforms: iOS, Android, WASI, and
+Emscripten. All of them will need to build packages via cross builds. Currently,
+``pyodide-build`` allows building a very large number of Python packages for
+Emscripten, but it is very complicated. Ideally, the Python packaging ecosystem
+would have standards for cross builds. This is a difficult long term project,
+particularly because the packaging system is complex and was designed from the
+ground up with the assumption that cross compilation would not happen.
```

Comment:
It's not a blocker on this PEP, but for public visibility - I'm hoping to kick start a discussion about this very shortly.